.github/workflows/spelling.yaml

```diff
@@ -0,0 +1,206 @@
+      if: env.MERGE_FAILED != '1'
+      id: spelling
+      uses: ./
+      with:
+        suppress_push_for_open_pull_request: 1
+        check_file_names: 1
+        experimental_path: introspected
+        spell_check_this: check-spelling/spell-check-this@prerelease
+        checkout: false
+        post_comment: 0
+        use_magic_file: 1
+        report-timing: 1
+        experimental_apply_changes_via_bot: 1
+        use_sarif: ${{ (!github.event.pull_request || (github.event.pull_request.head.repo.full_name == github.repository)) && 1 }}
+        extra_dictionary_limit: 10
+        # this repository doesn't use php, but the dictionary has fairly good coverage
+        extra_dictionaries:
+          cspell:software-terms/dict/softwareTerms.txt
+          cspell:php/dict/php.txt
+          cspell:node/node.txt
+          cspell:python/src/python/python-lib.txt
+        check_commit_messages: ${{ env.workflow_check_commit_messages }}
+        ignore-pattern: '[^\p{Ll}\p{Lm}\p{Lt}\p{Lu}]'
+        upper-pattern: '[\p{Lu}\p{Lt}\p{Lm}]'
+        lower-pattern: '[\p{Ll}\p{Lm}]'
+        not-lower-pattern: '[^\p{Ll}\p{Lm}]'
+        not-upper-or-lower-pattern: '[^\p{Lu}\p{Lt}\p{Lm}]'
+        punctuation-pattern: "'"
+
+  comment-push:
+    name: Report (Push)
+    # If your workflow isn't running on push, you can remove this job
+    runs-on: ubuntu-latest
+    needs: spelling
+    permissions:
+      contents: write
+    if: (success() || failure()) && needs.spelling.outputs.followup && github.event_name == 'push'
+    steps:
+    - name: checkout
+      uses: actions/checkout@v4
+    - name: comment
+      uses: ./
+      with:
+        checkout: false
+        spell_check_this: check-spelling/spell-check-this@prerelease
+        task: ${{ needs.spelling.outputs.followup }}
+        internal_state_directory: ${{ needs.spelling.outputs.docker_container && needs.spelling.outputs.internal_state_directory }}
+        caller_container: ${{ needs.spelling.outputs.docker_container }}
+
+  comment-pr:
+    name: Report (PR)
+    # If you workflow isn't running on pull_request*, you can remove this job
+    runs-on: ubuntu-latest
+    needs: spelling
+    permissions:
+      contents: read
+      pull-requests: write
+    if: (success() || failure()) && needs.spelling.outputs.followup && contains(github.event_name, 'pull_request')
+    steps:
+    - name: checkout
+      uses: actions/checkout@v4
+    - name: comment
+      uses: ./
+      with:
+        checkout: false
+        spell_check_this: check-spelling/spell-check-this@prerelease
+        task: ${{ needs.spelling.outputs.followup }}
+        internal_state_directory: ${{ needs.spelling.outputs.docker_container && needs.spelling.outputs.internal_state_directory }}
+        caller_container: ${{ needs.spelling.outputs.docker_container }}
+        experimental_apply_changes_via_bot: 1
+
+  update:
+    name: Update PR
+    permissions:
+      contents: write
+      pull-requests: write
```

Comment:
Does this job correct the spelling mistakes in the PR? If yes, then I think it is best to leave it to the contributor rather than updating it here. WDYT ?